### ISSUE TITLE
<ThemeProvider theme="primary"> does not apply theme class

### ISSUE THREAD

**Issue by yuchenshi** [jamesmfriedman/rmwc#567]
Sandbox: https://codesandbox.io/s/rmwc-sandbox-o3wwc

- What RMWC Version are you using [major.minor.patch]: 5.7.2 and 6.0.0-alpha.15
(See sandbox)

- Name your build system [Webpack, Rollup...]: Sandbox

- Describe the bug with as much detail as possible:

Try to render JSX like `<ThemeProvider options={{ primary: "#f00" }} theme="primary">foo</ThemeProvider>`

- What happened, and what was supposed to happen:

It actually generates `<div theme="primary">` in the DOM (note that the `theme` attribute is generated as-is without processing), which will have default text color in browser.

I would expect it to set the primary color to `#f00` and also apply `.mdc-theme-primary` class to the element (i.e. `<div class="mdc-theme-primary" style="...mdc-vars-here...">`), resulting in red text.

I really think rmwc should support this use case, but if not, I would expect this to be caught in TypeScript compile time (e.g. by removing the `theme` attribute from declared typing) rather than getting a div with a HTML `theme` attribute which does not affect theming at all.

**Comment by jamesmfriedman** [comment]
There is a theme component for applying specific colors and a ThemeProvider for specifying the values the variables are actually set to. Just switch to the Theme component.

**Comment by yuchenshi** [comment]
Thanks for the response. Yes, I understand the difference and the proposed solution.

For some background, I'm trying to write a component which needs to specify the variables for theming buttons within it, and the component itself also needs to have a themed surface background, so I wrote something like:

```jsx
<ThemeProvider options={...}>
  <Theme use="surface">
    {children}
  </Theme>
</ThemeProvider>
```

And then I wanted less nesting so I tried `<ThemeProvider options={...} theme="surface">` and I was surprised to learn it doesn't actually work, despite that type checker does not complain. Many other components accept a `theme` property so I assumed it will do the same thing here. I also understand that putting a `wrap` property will also work in terms of the final rendered DOM, but allowing `theme` on `<ThemeProvider>` would make the code more succinct. Thoughts?
